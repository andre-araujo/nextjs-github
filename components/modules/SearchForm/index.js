import { Component } from 'react';
import { func } from 'prop-types';

import Submit from '../../elements/Submit';
import TextInput from '../../elements/TextInput';

import {
    Form,
} from './styles';

class SearchForm extends Component {
    state = {}

    registerField = (proxyEvent) => {
        if (proxyEvent && proxyEvent.target && proxyEvent.target.name) {
            this.setState({
                [proxyEvent.target.name]: proxyEvent.target.value,
            });
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit(this.state);
    }

    render() {
        return (
            <Form
                name="search-form"
                onSubmit={this.handleSubmit}
            >
                <div>
                    <TextInput
                        name="user"
                        type="text"
                        onChange={this.registerField}
                    />
                </div>
                <Submit>
                    Buscar
                </Submit>
            </Form>
        );
    }
}

SearchForm.propTypes = {
    onSubmit: func.isRequired,
};

export default SearchForm;
