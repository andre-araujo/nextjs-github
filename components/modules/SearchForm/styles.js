import styled from 'styled-components';

export const Form = styled.form`
    max-width: 30rem;
    margin: 0 auto;
    padding: 1rem;
    text-align: center;
`;

export const Row = styled.div`
    margin-bottom: 2rem;
`;
