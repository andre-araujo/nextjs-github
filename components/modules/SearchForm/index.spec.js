import React from 'react';
import { mount } from 'enzyme';

import SearchForm from './index';

describe('SearchForm component', () => {
    it('should match snapshot', () => {
        expect(mount(
            <SearchForm
                onSubmit={() => {}}
            />,
        )).toMatchSnapshot();
    });

    it('should call onSubmit prop with input value', () => {
        const onSubmit = jest.fn();
        const searchForm = mount(
            <SearchForm
                onSubmit={onSubmit}
            />,
        );

        searchForm
            .find('input[name="user"]')
            .simulate('change', { target: { name: 'user', value: 'test' } });

        searchForm
            .find('form')
            .simulate('submit');

        expect(onSubmit).toHaveBeenCalledWith({ user: 'test' });
    });

    it('should return empty value onSubmit if input has no name', () => {
        const onSubmit = jest.fn();
        const searchForm = mount(
            <SearchForm
                onSubmit={onSubmit}
            />,
        );

        searchForm
            .find('input[name="user"]')
            .simulate('change', { target: { value: 'test' } });

        searchForm
            .find('form')
            .simulate('submit');

        expect(onSubmit).toHaveBeenCalledWith({});
    });
});
