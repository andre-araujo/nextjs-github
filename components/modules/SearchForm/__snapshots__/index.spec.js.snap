// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component should match snapshot 1`] = `
<form
  className="sc-EHOje bepPhO"
  name="search-form"
  onSubmit={[Function]}
>
  <div>
    <div
      className="sc-bwzfXH pHugw"
    >
      <div
        className="sc-htpNat iFhpIY"
      >
        <input
          className="sc-ifAKCX hpdfCg"
          name="user"
          onChange={[Function]}
          required={true}
          type="text"
        />
        <label
          className="sc-bxivhb frnCGp"
          htmlFor="user"
        >
          label
        </label>
      </div>
    </div>
  </div>
  <input
    className="sc-bdVaJa jDEzzO"
    type="submit"
    value="Buscar"
  />
</form>
`;
